### PR TITLE
Implement cov for ILMMs

### PR DESCRIPTION
**Pull request by willtebbutt**

### CODE REVIEW

**Review by thomasgudjonwright — COMMENTED** [review]
I am wondering which is more efficient:
1) Define `mean_and_cov` and `mean_and_var` then define `mean`, `cov` and `var` from these 
or
2) Define `mean`, `cov` and `var` then `mean_and_cov` and `mean_and_var` from those

I suppose internally right now we just use the `x_and_y` so likely we want those to be defined explicitly?

File: test/ilmm.jl [hunk 1]
```diff
@@ -31,8 +31,8 @@ function test_ilmm(rng, kernels, H, x_train, x_test, y_train, y_test)
     @test gradient(logpdf, pi, y_test) isa Tuple
 
     @testset "primary_public_interface" begin
-        test_finitegp_primary_public_interface(rng, ilmmx)
-        test_finitegp_primary_public_interface(rng, pi)
+        test_finitegp_primary_and_secondary_public_interface(rng, ilmmx)
```
[the remark this answers is not in the document]
Comment (willtebbutt) [reply test/ilmm.jl:34]: Hmm yeah, we ought to be able to compare against the naive KernelFunctions version I reckon. I'll add that.

**Comment by willtebbutt** [comment]
Not sure. Ideally we would define all of them separately (I think). I'll add the extra methods so that we know we're optimal.

File: test/ilmm.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (willtebbutt) [reply test/ilmm.jl:35]: Very glad you got me to do that. Found a bug!

**Comment by willtebbutt** [comment]
> Not sure. Ideally we would define all of them separately (I think). I'll add the extra methods so that we know we're optimal.

Hmm just looking at what it would take to do this, and I think it's going to add more code than I would like to add right now (I also think it's reasonable to consider `mean` and `var` out of scope for this PR). Would you mind if I open an issue about this so that we can get this functionality in?

**Review by thomasgudjonwright — APPROVED** [review]
LGTM! The issue you mentioned for me to make is [here] :)(https://github.com/invenia/LinearMixingModels.jl/issues/38).

**Pull request merged**